packages/flutter/test/material/carousel_theme_test.dart

```diff
@@ -0,0 +1,216 @@
+            ),
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    expect(find.byType(CarouselView), findsOneWidget);
+
+    final Finder padding = find.descendant(
+      of: find.byType(CarouselView),
+      matching: find.byWidgetPredicate(
+        (Widget widget) => widget is Padding && widget.child is Material,
+      ),
+    );
+
+    expect(padding, findsOneWidget);
+    final Padding paddingWidget = tester.widget<Padding>(padding);
+    final Material material = paddingWidget.child! as Material;
+
+    final InkWell inkWell = tester.widget<InkWell>(
+      find.descendant(of: find.byType(CarouselView), matching: find.byType(InkWell)),
+    );
+
+    expect(paddingWidget.padding, carouselViewTheme.padding);
+    expect(material.color, carouselViewTheme.backgroundColor);
+    expect(material.elevation, carouselViewTheme.elevation);
+    expect(material.shape, carouselViewTheme.shape);
+    expect(inkWell.overlayColor, carouselViewTheme.overlayColor);
+  });
+
+  testWidgets('Widgets properties override theme', (WidgetTester tester) async {
+    final CarouselViewThemeData carouselViewTheme = _carouselViewThemeData();
+    const Color backgroundColor = Color(0xFFFF0000);
+    const double elevation = 10.0;
+    const EdgeInsets padding = EdgeInsets.all(15.0);
+    const OutlinedBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(10)),
+    );
+    const WidgetStateProperty<Color?> overlayColor = MaterialStatePropertyAll<Color>(Colors.green);
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(carouselViewTheme: carouselViewTheme),
+        home: const Scaffold(
+          body: Center(
+            child: CarouselView(
+              backgroundColor: backgroundColor,
+              elevation: elevation,
+              padding: padding,
+              shape: shape,
+              overlayColor: overlayColor,
+              itemExtent: 100,
+              children: <Widget>[SizedBox(width: 100, height: 100)],
+            ),
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    expect(find.byType(CarouselView), findsOneWidget);
+
+    final Finder paddingFinder = find.descendant(
+      of: find.byType(CarouselView),
+      matching: find.byWidgetPredicate(
+        (Widget widget) => widget is Padding && widget.child is Material,
+      ),
+    );
+
+    expect(paddingFinder, findsOneWidget);
+    final Padding paddingWidget = tester.widget<Padding>(paddingFinder);
+    final Material material = paddingWidget.child! as Material;
+
+    final InkWell inkWell = tester.widget<InkWell>(
+      find.descendant(of: find.byType(CarouselView), matching: find.byType(InkWell)),
+    );
+
+    expect(paddingWidget.padding, padding);
+    expect(material.color, backgroundColor);
+    expect(material.elevation, elevation);
+    expect(material.shape, shape);
+    expect(inkWell.overlayColor, overlayColor);
+  });
+}
+
```

Comment:
Can we also test local `CarouselViewTheme` can override `Theme.carouselViewTheme`?